app/src/main/java/infrmr/newsapp/github/com/ifrmr/ArticleListFragment.java

```diff
@@ -0,0 +1,135 @@
+package infrmr.newsapp.github.com.ifrmr;
+
+import android.app.Activity;
+import android.content.res.Configuration;
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.support.v7.widget.GridLayoutManager;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+/**
+ * A fragment representing a list of Items. Large screen devices (such as tablets) are supported by replacing the ListView
+ * with a GridView. Activities containing this fragment MUST implement the {@link OnFragmentInteractionListener}
+ * interface.
+ */
+public class ArticleListFragment extends Fragment {
+
+    // Adapter for RecyclerView
+    private RecyclerView.Adapter mAdapter;
+    // Activity callback listener
+    private OnFragmentInteractionListener mListener;
+
+    /**
+     * The fragment argument representing the section number for this
+     * fragment.
+     */
+    private static final String ARG_SECTION_NUMBER = "section_number";
+
+    /**
+     * Boolean to detect whether the fragment is still visable
+     */
+    public static boolean isVisable;
+
+    /**
+     * Mandatory empty constructor for the fragment manager to instantiate the
+     * fragment (e.g. upon screen orientation changes).
+     */
+    public ArticleListFragment() {
+    }
+
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.recycler_view_layout, container, false);
+
+        // Find and init RecyclerView
+        RecyclerView mRecyclerView = (RecyclerView) view.findViewById(R.id.recycleView);
+        //mRecyclerView.setHasFixedSize(true);
+
+        // If big screen, use grid layout, else use small
+        if ((getResources().getConfiguration().screenLayout &
+                Configuration.SCREENLAYOUT_SIZE_MASK) ==
+                Configuration.SCREENLAYOUT_SIZE_LARGE) {
+            mRecyclerView.setLayoutManager(new GridLayoutManager(getActivity(), 2));
+        } else {
+            LinearLayoutManager linearLayoutManager = new LinearLayoutManager(getActivity());
+            linearLayoutManager.setOrientation(LinearLayoutManager.VERTICAL);
+            mRecyclerView.setLayoutManager(linearLayoutManager);
+        }
+
+        // specify our custom adapter
+        mAdapter = new RecyclerAdapter(getActivity());
+        mRecyclerView.setAdapter(mAdapter);
+
+        return view;
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        isVisable = true;
```

Comment:
Avoid using variables to track lifecycle states, its redundant and error prone. Android tracks most of the states...try getView(), might be the thing you need here.